math/mathcore/inc/TMath.h

```diff
@@ -1463,9 +1463,11 @@ template <typename T> Double_t TMath::ModeHalfSample(Long64_t n, const T *a, con
       const auto vstop = values.end();
       const auto viter = std::lower_bound(vstart, vstop, value);
       const auto vidx = std::distance(vstart, viter);
-      if (viter != vstop && w) {
+      if (viter != vstop && *viter == value) {
```

Comment:
```suggestion
      if (w && viter != vstop && *viter == value) {
```
Thanks.
However: I think that deduplication must be only done if weights vary.
Why? The Bickel algorithm expects all values have the same weight of 1. 
The deduplication is just a fallback when the user passes varying weights, then we fall back to getting the Mode of the sample.
Otherwise, we apply the Bickel algorithm.

Reply:
I have updated the PR.